feed-rs/src/xml/mod.rs

```diff
@@ -483,32 +490,36 @@ impl XmlEvent {
     // Creates a new event corresponding to an XML end-tag
     fn end<R: BufRead>(event: &BytesEnd, reader: &Reader<R>) -> XmlEvent {
         // Parse the name
-        let name = XmlEvent::parse_name(event.name(), reader);
+        let name = XmlEvent::parse_name(event.name().as_ref(), reader);
 
         XmlEvent::End { name }
     }
 
     // Extracts the element name, dropping the namespace prefix if present
     fn parse_name<R: BufRead>(bytes: &[u8], reader: &Reader<R>) -> String {
-        reader.decode(bytes).split(':').rev().next().unwrap_or("").into()
+        reader
+            .decoder()
+            .decode(bytes)
+            .ok()
+            .and_then(|name| name.split(':').rev().next().map(str::to_string))
+            .unwrap_or_default()
     }
 
     // Creates a new event corresponding to an XML start-tag
     fn start<R: BufRead>(namespace: NS, event: &BytesStart, reader: &Reader<R>) -> XmlEvent {
         // Parse the name
-        let name = XmlEvent::parse_name(event.name(), reader);
+        let name = XmlEvent::parse_name(event.name().as_ref(), reader);
 
         // Parse the attributes
         let attributes = event
             .attributes()
             .filter_map(|a| {
                 if let Ok(a) = a {
-                    let name = reader.decode(a.key);
+                    let name = reader.decoder().decode(a.key.as_ref()).unwrap();
```

Comment:
This needs to be fixed as this shouldn't panic. Seen as this is a `filter_map` it might be acceptable to return `None` here and skip the attribute, but I don't know enough about the logic here to make that determination.

Reply:
I've pushed another commit that replaces these `.unwrap()`s with `return None` to skip invalid attributes.

Reply:
Great, thanks!